### ISSUE TITLE
Remote backend with HTTP connector getting stuck on a loop

### ISSUE THREAD

**Issue by fermayo** [PowerDNS/pdns#2069]
Hello,

I'm using the remote backend on pdns authoritative server 3.4.1 with an HTTP connector (non-SSL). From time to time, pdns CPU load goes to 100% with the following strace:
```
[pid 12075] poll([{fd=11, events=POLLIN}], 1, 2000) = 1 ([{fd=11, revents=POLLIN}])
[pid 12075] recvfrom(11, "", 4096, 0, NULL, NULL) = 0
[pid 12075] poll([{fd=11, events=POLLIN}], 1, 2000) = 1 ([{fd=11, revents=POLLIN}])
[pid 12075] recvfrom(11, "", 4096, 0, NULL, NULL) = 0
[...]
```
And gets stuck there forever until restarted.

Any ideas?

Thanks,
Fernando.

**Comment by cmouse** [comment]
I'll look into it.

**Comment by cmouse** [comment]
Can you compile test from branch and see if it helps? https://github.com/cmouse/pdns/tree/httpconnector-timeout

**Comment by fermayo** [comment]
OK, let me build it and test it. Thanks!

**Comment by fermayo** [comment]
Now I'm getting this:
```
Jan 13 13:18:50 While reading from HTTP endpoint 10.0.0.41:80: timeout reading
Jan 13 13:28:54 While reading from HTTP endpoint 10.0.1.183:80: timeout reading
Jan 13 13:36:44 While reading from HTTP endpoint 10.0.1.183:80: timeout reading
Jan 13 14:16:24 While reading from HTTP endpoint 10.0.1.183:80: timeout reading
Jan 13 14:16:24 While reading from HTTP endpoint 10.0.0.41:80: timeout reading
Jan 13 14:17:52 While reading from HTTP endpoint 10.0.1.183:80: timeout
Jan 13 14:33:19 While reading from HTTP endpoint 10.0.1.183:80: timeout reading
```

And CPU is stable. Is that working then?

**Comment by cmouse** [comment]
Do you get replies?


---
Aki Tuomi


-------- Original message --------

**Comment by fermayo** [comment]
Yes, seems to be replying to DNS queries

**Comment by cmouse** [comment]
The error means you are not replying with valid http response. Bug on our side was not dealing with this. 


---
Aki Tuomi


-------- Original message --------

**Comment by cmouse** [comment]
I'll provide a better fix bit later. This was debug version.

**Comment by fermayo** [comment]
Looking forward to it. Thanks!

**Comment by cmouse** [comment]
Should be ok now

**Comment by fermayo** [comment]
Thanks!

**Comment by cmouse** [comment]
@fermayo it seems I did not fix it correctly after all since I misunderstood sstuff.hh, there is additional fixes that will make it wait less time, see #2086
